urdf/psen_scan.urdf.xacro

```diff
@@ -0,0 +1,102 @@
+    <xacro:property name="max_range" value="4.0" /> <!-- TODO correct dimensions -->
+    <xacro:property name="min_angle" value="$(eval -137.5 /180.0 * pi)" /> <!-- src datasheet p 91 -->
+    <xacro:property name="max_angle" value="$(eval 137.5 /180.0 * pi)" /> <!-- src datasheet p 91 -->
+
+    <xacro:property name="mass" value="1.53" /> <!-- src datasheet p 96 -->
+    <xacro:property name="radius" value="0.051" /> <!-- src datasheet p 96 -->
+    <xacro:property name="height" value="0.152" /> <!-- src datasheet p 96 -->
+    <xacro:property name="dist_scanning_plane_top" value="0.0377" /> <!-- src datasheet p 96 -->
+    <xacro:property name="dist_scanning_axis_rear" value="0.0525" /> <!-- src datasheet p 96 -->
+        
+    <!-- Location of mounting point at the lower back of device -->
+    <link name="${prefix}_mount_link">
+      <inertial>
+        <mass value="0.001" />
+        <origin xyz="0 0 0" />
+        <inertia ixx="0.000001" ixy="0.0" ixz="0.0"
+          iyy="0.000001" iyz="0.0" izz="0.000001" /> <!-- TODO correct dimensions -->
+      </inertial>
+    </link>
+
+    <joint name="${prefix}_mount_joint" type="fixed">
+      <parent link="${prefix}_mount_link" />
+      <child link="${prefix}" />
+      <origin rpy="0 0 0" xyz="${radius} 0 ${height / 2}"/>
+      
+    </joint>
+
+    <!-- Location of scanner body -->
+    <link name="${prefix}">
+      <visual>
+        <geometry>
+            <cylinder radius="${radius}" length="${height}" />
+        </geometry>
+        <material name="pilz_yellow" >
+          <color rgba="1.0 0.722 0.196 1.0" />
+        </material>
+      </visual>
+      <collision>
+        <geometry>
+            <cylinder radius="${radius}" length="${height}" />
+        </geometry>
+      </collision>
+      <inertial>
+        <mass value="${mass}" />
+        <origin xyz="0 0 0" />
+        <inertia ixx="${0.0833333 * mass * (radius * radius + height * height)}" ixy="0.0" ixz="0.0"
+          iyy="${0.0833333 * mass * (radius * radius + height * height)}" iyz="0.0"
+          izz="${0.0833333 * mass * (radius * radius + radius * radius)}" /> <!-- TODO correct dimensions -->
+      </inertial>
+    </link>
+    <xacro:psen_scan_gazebo_v0 name="${prefix}" link="${prefix}" ros_topic="${ros_topic}" update_rate="15.0" min_angle="${min_angle}" max_angle="${max_angle}" min_range="${min_range}" max_range="${max_range}"/>
+
+    <joint name="${prefix}_scan_joint" type="fixed">
+      <parent link="${prefix}" />
+      <child link="${prefix}_scan" />
+      <origin rpy="${pi} 0 0" xyz="0 ${radius - dist_scanning_axis_rear} ${height / 2 - dist_scanning_plane_top}"/>
+    </joint>
+    
+    <!-- This is the location of the laser scan -->
+    <link name="${prefix}_scan">
+    </link>
+  </xacro:macro>
+
+
+  <xacro:macro name="psen_scan_gazebo_v0" params="name link ros_topic update_rate min_angle max_angle min_range max_range">
```

Comment:
Do we need this?

Reply:
not at the moment. I think later we should definitely also think about how to best help costumers with their gazebo model. And this might just be it. But for now, it is not necessary.